Codeinwp/neve
# feat: added better support for keyboard menu navigation

### Summary

Added better support for keyboard navigation.
I moved the sidebar animation inline and is enqueued only if sidebar navigation is used.
Fixed JS trap for sidebar to allow removing trap on menu close. Previously was only working on ESC.

For mega menu use this PR for Neve Pro:
https://github.com/Codeinwp/neve-pro-addon/pull/2268

Previous behavior:
Before it would open the sub-menu automatically on parent TAB.
Then it would TAB through all children before moving to the next item.

Current behavior:
Now the caret is a focus-able element and will accept ENTER as a key to toggle the sub-menu
TAB only cycles through visible/opened items.
If the sub-menu is not opened from the caret, the TAB will move to next item.


### Will affect visual aspect of the product

NO

### Screenshots 
https://d.pr/v/NUoLv6


### Test instructions

1. Check with a menu that has sub-menus 
2. Use the keyboard to navigate
3. Ensure that all previous functionality behaves the same.


Closes #2935.
Closes #3731.
